oak-blob-plugins/src/main/java/org/apache/jackrabbit/oak/plugins/blob/datastore/FSBackend.java

```diff
@@ -105,20 +105,28 @@ public InputStream read(DataIdentifier identifier) throws DataStoreException {
     @Override
     public void write(DataIdentifier identifier, File file) throws DataStoreException {
         File dest = getFile(identifier, fsPathDir);
-        synchronized (this) {
-            if (dest.exists()) {
-                long now = System.currentTimeMillis();
-                if (getLastModified(dest) < now + ACCESS_TIME_RESOLUTION) {
-                    setLastModified(dest, now + ACCESS_TIME_RESOLUTION);
-                }
-            } else {
-                try {
-                    FileUtils.copyFile(file, dest);
-                } catch (IOException ie) {
-                    LOG.error("failed to copy [{}] to [{}]", file.getAbsolutePath(),
-                        dest.getAbsolutePath());
-                    throw new DataStoreException("Not able to write file [" + identifier + "]", ie);
+        if (dest.exists()) {
```

Comment:
What is required is to copy the file already created in the caching layer (which is local fs) to a temp file in NFS (which the FSBackend would typically represent) and then do everything in the synchronized block.